packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanelMain.tsx

```diff
@@ -794,29 +853,38 @@ export function TransferPanelMain({
         />
       </div>
 
-      <NetworkContainer network={to}>
+      <NetworkContainer
+        network={to}
+        balanceFor={
+          walletAddress && walletAddress.toLowerCase() !== destinationAddress
+            ? destinationAddress
+            : undefined
+        }
+      >
         <NetworkListboxPlusBalancesContainer>
           <NetworkListbox label="To:" {...networkListboxProps.to} />
           <BalancesContainer>
             {isSwitchingL2Chain ? (
               <StyledLoader />
             ) : (
-              <>
-                <TokenBalance
-                  balance={
-                    app.isDepositMode
-                      ? selectedTokenBalances.l2
-                      : selectedTokenBalances.l1
-                  }
-                  on={app.isDepositMode ? NetworkType.l2 : NetworkType.l1}
-                  forToken={selectedToken}
-                  prefix={selectedToken ? 'BALANCE: ' : ''}
-                />
-                <ETHBalance
-                  balance={app.isDepositMode ? ethL2Balance : ethL1Balance}
-                  prefix={selectedToken ? '' : 'BALANCE: '}
-                />
-              </>
+              utils.isAddress(destinationAddressOrWalletAddress) && (
```

Comment:
To me it makes most sense to hide balances if the destination address is invalid. Better than having an endless spinner.

<img width="526" alt="Screenshot 2023-06-20 at 17 43 18" src="https://github.com/OffchainLabs/arbitrum-token-bridge/assets/47932951/6eac6674-3bea-
482c-ac17-d2f839de3a75">